components/InputAmount.tsx

```diff
@@ -24,12 +24,25 @@ const formatCurrencyName = (currency, currencyDisplay) => {
 };
 
 const parseValueFromEvent = (e, precision, ignoreComma = false) => {
-  if (e.target.value === '') {
+  let value = e.target.value as string;
+  if (value === '') {
     return null;
-  } else {
-    const parsedNumber = parseFloat(ignoreComma ? e.target.value.replace(',', '') : e.target.value);
-    return isNaN(parsedNumber) ? NaN : parsedNumber.toFixed(precision);
   }
+
+  if (value.endsWith(',') || value.endsWith('.')) {
+    value += '0';
+  }
+
+  if (value.startsWith(',') || value.startsWith('.')) {
+    value = `0${value}`;
+  }
+
+  if (ignoreComma) {
+    value = value.replaceAll(',', '.');
```

Comment:
This shortcut could trigger issues with people pasting or typing numbers like `1,234.42`.

I tried to look around at what others are doing, and it seems that a common pattern is to look at what's the locale character for thousands separators and completely ignore (remove) them. Example: https://design.pega.com/design/currency-input

If that's too complex or if it introduces other issues, I'm ok with the current approach, but in that case maybe try to handle the case when pasting a number with both comma and dot (only consider the last separator).